### PR TITLE
add pinot.minion prefix on minion configs for consistency

### PR DESCRIPTION
**Pull request by klsince**
## Description
added `pinot.minion.` prefix on minion configs for consistency, but keeping the old config names for backward compatibility. 

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java [hunk 1]
```diff
@@ -453,10 +453,16 @@ private CommonConstants() {
...
-    public static final String PREFIX_OF_CONFIG_OF_SEGMENT_FETCHER_FACTORY = "segment.fetcher";
-    public static final String PREFIX_OF_CONFIG_OF_SEGMENT_UPLOADER = "segment.uploader";
-    public static final String PREFIX_OF_CONFIG_OF_PINOT_CRYPTER = "crypter";
+
+    // Add pinot.minion prefix on those configs to be consistent with configs of controller and server.
+    public static final String PREFIX_OF_CONFIG_OF_PINOT_FS_FACTORY = "pinot.minion.storage.factory";
+    public static final String PREFIX_OF_CONFIG_OF_SEGMENT_FETCHER_FACTORY = "pinot.minion.segment.fetcher";
+    public static final String PREFIX_OF_CONFIG_OF_SEGMENT_UPLOADER = "pinot.minion.segment.uploader";
+    public static final String PREFIX_OF_CONFIG_OF_PINOT_CRYPTER = "pinot.minion.crypter";
+    public static final String DEPRECATED_PREFIX_OF_CONFIG_OF_PINOT_FS_FACTORY = "storage.factory";
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:462]: Please add deprecated annotation to these keys

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

**Review by Jackie-Jiang — COMMENTED** [review]
`CONFIG_OF_METRICS_PREFIX_KEY` and `CONFIG_OF_TASK_AUTH_TOKEN` should also be changed

**Comment by klsince** [comment]
> `CONFIG_OF_METRICS_PREFIX_KEY` and `CONFIG_OF_TASK_AUTH_TOKEN` should also be changed

updated CONFIG_OF_METRICS_PREFIX_KEY. 

CONFIG_OF_TASK_AUTH_TOKEN is actually consistent between server/minion, but w/o component prefix. lemme know if that's concerning and will update.

**Comment by Jackie-Jiang** [comment]
> CONFIG_OF_TASK_AUTH_TOKEN is actually consistent between server/minion, but w/o component prefix. lemme know if that's concerning and will update.

Sure, we can fix it separately

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**